user-org-keycloak-util/src/main/java/org/sunbird/sso/service/impl/KeyCloakRsaKeyFetcher.java

```diff
@@ -0,0 +1,133 @@
+package org.sunbird.sso.service.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.KeyFactory;
+import java.security.PublicKey;
+import java.security.spec.RSAPublicKeySpec;
+import java.util.Base64;
+import java.util.Base64.Decoder;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.sunbird.sso.util.SSOConstant;
+import org.sunbird.util.LoggerEnum;
+import org.sunbird.util.ProjectLogger;
+import org.sunbird.util.PropertiesCache;
+
+/** Class to fetch SSO public key from Keycloak server using 'certs' HTTP API call. */
+public class KeyCloakRsaKeyFetcher {
+  private static final String MODULUS = "modulusBase64";
+  private static final String EXPONENT = "exponentBase64";
+
+  /**
+   * This method will accept keycloak base URL and realm name. Based on provided values it will
+   * fetch public key from keycloak.
+   *
+   * @param url A string value having keycloak base URL
+   * @param realm Keycloak realm name
+   * @return Public key used to verify user access token.
+   */
+  public PublicKey getPublicKeyFromKeyCloak(String url, String realm) {
+    try {
+      Map<String, String> valueMap = null;
+      Decoder urlDecoder = Base64.getUrlDecoder();
+      KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+      String publicKeyString = requestKeyFromKeycloak(url, realm);
+      if (publicKeyString != null) {
+        valueMap = getValuesFromJson(publicKeyString);
+        if (valueMap != null) {
```

Comment:
Use MapUtils.isNotEmpty

Reply:
to use map utils i need to include appache common collection4 jar.